R/cor_test_shepherd.R

```diff
@@ -16,3 +16,84 @@
   out$Method <- "Shepherd's Pi"
   out
 }
+
+
+# Utils -------------------------------------------------------------------
+
+#' @keywords internal
+.robust_bootstrap_mahalanobis <- function(data, iterations = 1000) {
+  Ms <- replicate(n = iterations, {
+    # Draw random numbers from 1:n with replacement
+    idx <- sample(nrow(data), replace = TRUE)
+    # Resample data
+    dat <- data[idx, ]
+    # Calculating the Mahalanobis distance for each actual observation using resampled data
+    stats::mahalanobis(data, center = colMeans(dat), cov = stats::cov(dat))
+  })
+
+  apply(Ms, 1, median)
+}
+
+
+# # Mahalanobis distance and confidence interval (CI)
```

Comment:
Why are we commenting this code and not just outright removing it?

Reply:
I feel bad deleting Dom's code...